### PR TITLE
Timespinner - Add Lore Checks checks

### PR DESCRIPTION
**Pull request by TriumphantBass**
Adds checks associated with the "Lore Checks" flag in Timespinner.

Companion commit merged to Timespinner Randomizer is https://github.com/JarnoWesthof/TsRandomizer/commit/776bd3743df823f13814efdf9e7123045a29184e

### CODE REVIEW

**Comment by TriumphantBass** [comment]
Brought to my attention I need to update the data version in init, will do that shortly

File: worlds/timespinner/Locations.py [hunk 1]
```diff
@@ -217,7 +217,34 @@ def get_locations(world: Optional[MultiWorld], player: Optional[int]) -> Tuple[L
...
+            LocationData('Forest', 'Journal - Forest Bat Jump Ledge (Peace Treaty)',  1337189),
+            LocationData('Castle Ramparts', 'Journal - Floating in Moat (Prime Edicts)',  1337190),
+            LocationData('Castle Ramparts', 'Journal - Archer + Knight (Declaration of Independence)',  1337191),
+            LocationData('Castle Keep', 'Journal - Under the Twins (Letter of Reference)',  1337192),
+            LocationData('Castle Keep', 'Journal - Castle Loop Giantess (Political Advice)',  1337193),
+            LocationData('Castle Keep', 'Journal - Aleana\'s Room (Diplomatic Missive)',  1337194),
+            LocationData('Royal Towers', 'Journal - Top Struggle Juggle Base (War of the Sisters)',  1337195),
+            LocationData('Royal Towers', 'Journal - Aleana Boss (Stained Letter)',  1337196),
+            LocationData('Royal Towers', 'Journal - Near Bottom Struggle Juggle (Mission Findings)',  1337197),
+            LocationData('Caves of Banishment (Maw)', 'Journal - Lower Left Maw Caves (Naivety)',  1337198),
```
Comment (JarnoWesthof) [inline worlds/timespinner/Locations.py:244]: , at the end of the last line is not needed

File: worlds/timespinner/Locations.py [hunk 2]
```diff
@@ -217,7 +217,34 @@ def get_locations(world: Optional[MultiWorld], player: Optional[int]) -> Tuple[L
             LocationData('Left Side forest Caves', 'Cantoran',  1337176),
         )
 
-    # 1337177 - 1337236 Reserved for future use
+    # 1337177 - 1337198 Lore Checks
```
Comment (JarnoWesthof) [inline worlds/timespinner/Locations.py:220]: Adding locations also required an update of data_version in _init_.py

Some locations do need thier requirements added, so a requirement is based on its region which has requirements to reach a certain region + additional requirement for certain checks like the breakable wall in the lab for the memories

**Review by JarnoWesthof — CHANGES_REQUESTED** [review]

**Comment by TriumphantBass** [comment]
Closing until I can address these

File: worlds/timespinner/Locations.py (hunk 2 above)
Comment (TriumphantBass) [reply worlds/timespinner/Locations.py:220]: Added

File: worlds/timespinner/Locations.py (hunk 1 above)
Comment (TriumphantBass) [reply worlds/timespinner/Locations.py]: Removed

**Comment by Berserker66** [comment]
Thank you!

**Pull request merged**